server/cmd/dcrdex/log.go

```diff
@@ -50,61 +50,58 @@ var (
 	// It should be closed on application shutdown.
 	logRotator *rotator.Rotator
 
-	// lm is used to create dex.Loggers for all DEX subsystems. Loggers must
-	// not be used before the log rotator has been initialized, or data
-	// races and/or nil pointer dereferences will occur.
-	lm *dex.LoggerMaker
-
 	// package main's Logger
 	log dex.Logger
 
 	// subsystemLoggers maps each subsystem identifier to its associated logger.
 	subsystemLoggers map[string]dex.Logger
+
+	// The default log level for the pre-defined package level loggers. This
+	// should correspond to defaultLogLevel, but there is no map to ensure it.
+	defaultLevel = dex.LevelDebug
 )
 
 func init() {
-	// lm is used to create dex.Loggers for all DEX subsystems. Loggers must
-	// not be used before the log rotator has been initialized, or data
-	// races and/or nil pointer dereferences will occur.
-	var err error
-	lm, err = dex.NewLoggerMaker(logWriter{}, defaultLogLevel)
-	if err != nil {
-		panic(err)
+	// Loggers must not be used before the log rotator has been initialized, or
+	// data races and/or nil pointer dereferences will occur.
+
+	makeLogger := func(name string) dex.Logger {
+		return dex.NewLogger(name, defaultLevel, logWriter{})
 	}
```

Comment:
Just noticing that we're setting these log levels to default, only to reset them in `parseAndSetDebugLevels`. Any reason we can't just move this `init` body to `parseAndSetDebugLevels`? We could then drop [this block](https://github.com/decred/dcrdex/blob/e4a9613f1f45a471026e4b0a1ce70ac0de3ed189/server/cmd/dcrdex/config.go#L204-L215), since the `LoggerMaker` takes care of it based on the `debugLevel` string. We're not using logging before a successful `loadConfig` anyway.

Reply:
Will try that out.  I wish I had split this logging fix out because it's getting hairy, or at least I'm having difficulty wrapping my head around it now for some reason.

Reply:
That turned out to be tricky, but I made the change.  We can't have multiple slog.Backends using the same logrotator because logrotator.Write isn't safe for concurrent access.  This prevents the `log` variable from being ready at startup.  This will almost certainly need to be reworked again as it's terribly hard to follow and full of limitations, but I really am sick of messing with the loggers.